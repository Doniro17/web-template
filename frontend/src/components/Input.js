import React from 'react';

export default function Input({
  type,
  placeholder,
  className,
  onChange,
  value,
  name,
  id,
  label,
  labelStyle,
}) {
  return (
    <label className={labelStyle} htmlFor={id}>
      {label}
      <input
        id={id}
        name={name}
        type={type}
        placeholder={placeholder}
        className={className}
        onChange={onChange}
        value={value}
      />
      {type === 'checkbox' && <span className="checkmark" />}
    </label>
  );
}
