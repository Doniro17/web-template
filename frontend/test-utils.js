import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
