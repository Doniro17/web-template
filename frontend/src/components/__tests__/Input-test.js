import { render } from 'test-utils';
import Input from '../Input';

const testValue = '';
const props = {
  label: 'Client',
  labelStyle: 'labelStyle',
  type: 'checkbox',
  value: testValue,
  onChange: jest.fn(),
  className: 'testClass',
  name: 'testName',
};

describe('Input', () => {
  it('renders correctly', () => {
    const { container } = render(<Input {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
