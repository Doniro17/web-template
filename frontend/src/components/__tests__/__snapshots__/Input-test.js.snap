// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<label
  class="labelStyle"
>
  Client
  <input
    class="testClass"
    name="testName"
    type="checkbox"
    value=""
  />
  <span
    class="checkmark"
  />
</label>
`;
